### DISCUSSION TITLE
Please explain to me some basics about how lift/retraction distance and speeds are being set

### DISCUSSION THREAD

**Discussion by clytle374** [sn4k3/UVtools#969] [General]
I'm brand new to resin printing. Used FDM for years, and prusaslicer since it came out.  My printer is a Uniformation GKtwo, prusaslicer 2.8 and UVtool 5.0.0 if that matters.

I loaded the profiles into prusaslicer with UVTool and for the most part everything just works, but then I tried to print some parts flat on the build plate.  It appears the lift speed and distances are being ignored when they are set one this page in the following screenshot, Also a minimum of 1mm/s is 60mm/min and already on the high side for a minimum value.  
![Screenshot_20241222_203631](https://github.com/user-attachments/assets/4d95cd6c-bf6c-471c-976b-8a80e64f2345)

But then I see this.  Is this where these settings are coming from?  At least in FDM world I could read gcode files since I was a CNC person in a previous life.  
```

Don't remove the following keywords! These keywords are used in the "compatible printer" condition of the print and filament profiles to link the particular print and filament profiles to this printer profile.
PRINTER_VENDOR_PRUSA3D
PRINTER_MODEL_SL1
PRINTER_VENDOR_UNIFORMATION
PRINTER_MODEL_GKTWO
FILEFORMAT_GKTWO.CTB
FILEVERSION_4

START_CUSTOM_VALUES
BottomWaitTimeBeforeCure_8
WaitTimeBeforeCure_1.5
BottomWaitTimeAfterCure_3
WaitTimeAfterCure_1
BottomLiftHeight_5
LiftHeight_5
BottomLiftSpeed_80
LiftSpeed_150
BottomWaitTimeAfterLift_0
WaitTimeAfterLift_0
RetractSpeed_150
END_CUSTOM_VALUES
```
Is this where I need to change the settings.  Sorry for newbe question.
Thanks! Cory

**Comment by sn4k3** [comment]
The lifts must be set under Printer notes, after `START_CUSTOM_VALUES`
The reason is PrusaSlicer didn't have and still don't have most of the values other printers uses, so they are set under notes.
The keyword list can be seen here: https://github.com/sn4k3/UVtools?tab=readme-ov-file#prusaslicer

When open in UVtools you can confirm the actual values on status bar

**Reply by clytle374** [reply]
Thank you.  In hindsight the readme makes perfect sense now.  But that didn't click at first.
One more question.  Is there an order of precidence for these keywords?  Example: I want a default in printer notes but want to override on for one material, can I override printer notes with material notes?

**Reply by sn4k3** [reply]
Order doesn't matter.
You can override them per material in it notes as you suggest. Material notes have priority.
